clippy_lints/src/non_canonical_impls.rs

```diff
@@ -112,140 +112,142 @@ declare_clippy_lint! {
 declare_lint_pass!(NonCanonicalImpls => [NON_CANONICAL_CLONE_IMPL, NON_CANONICAL_PARTIAL_ORD_IMPL]);
 
 impl LateLintPass<'_> for NonCanonicalImpls {
-    #[expect(clippy::too_many_lines)]
     fn check_impl_item<'tcx>(&mut self, cx: &LateContext<'tcx>, impl_item: &ImplItem<'tcx>) {
-        let Node::Item(item) = cx.tcx.parent_hir_node(impl_item.hir_id()) else {
-            return;
-        };
-        let Some(trait_impl) = cx.tcx.impl_trait_ref(item.owner_id).map(EarlyBinder::skip_binder) else {
-            return;
-        };
-        if cx.tcx.is_automatically_derived(item.owner_id.to_def_id()) {
-            return;
-        }
-        let ImplItemKind::Fn(_, impl_item_id) = cx.tcx.hir_impl_item(impl_item.impl_item_id()).kind else {
-            return;
-        };
-        let body = cx.tcx.hir_body(impl_item_id);
-        let ExprKind::Block(block, ..) = body.value.kind else {
-            return;
-        };
-        if block.span.in_external_macro(cx.sess().source_map()) || is_from_proc_macro(cx, impl_item) {
-            return;
+        if let Node::Item(item) = cx.tcx.parent_hir_node(impl_item.hir_id())
+            && let Some(trait_impl) = cx.tcx.impl_trait_ref(item.owner_id).map(EarlyBinder::skip_binder)
+            && !cx.tcx.is_automatically_derived(item.owner_id.to_def_id())
+            && let ImplItemKind::Fn(_, impl_item_id) = cx.tcx.hir_impl_item(impl_item.impl_item_id()).kind
+            && let body = cx.tcx.hir_body(impl_item_id)
+            && let ExprKind::Block(block, ..) = body.value.kind
+            && !block.span.in_external_macro(cx.sess().source_map())
+            && !is_from_proc_macro(cx, impl_item)
+        {
+            let trait_name = cx.tcx.get_diagnostic_name(trait_impl.def_id);
```

Comment:
Move this check into a fairly up position with `matches!(trait_name, Some(sym::Clone) | Some(sym::PartialOrd))`, as it's a [fairly cheap check](https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_middle/ty/context.rs.html#1841-1843)